### DISCUSSION TITLE
Question: Multiple Rate Limit ?

### DISCUSSION THREAD

**Discussion by bricemaurin** [OptimalBits/bull#2114] [Q&A]
Hey Bull Team 👋 
Thanks for the awesome work you've done with Bull, this is by far the best JS queue out there.

We implemented it and can't seem to find the answer about a specific point in the doc. I hope someone will be able to point me to the right spec / code to better understand this.

We use a rate limited api that has both a per minute, per hour and per day limit. How to implement that using Bull ?

Concretely, we can't call this API more than
- 3x per second
- 100x per minutes
- 300x per hour
- 2000x per day

How should we set this ?

Thanks !

**Comment by manast** [comment]
It is not possible to set more than one rate limiter per queue. Maybe a solution is to chain several queues with different rate limits, from the least restrictive to the most restrictive. So the first queue just adds the job to the next queue and so on, and the last queue performs the task.

**Reply by bricemaurin** [reply]
OK, thanks for your answer @manast !

**Reply by manast** [reply]
I am not sure what you are trying to accomplish with the code above. I see that you are using the groupKey, maybe you should start with a plain rate limiter first and then if you need rate limiting by groups I heavily recommend you to use the Pro version: https://docs.bullmq.io/bullmq-pro/groups/rate-limiting

**Reply by msalvatti** [reply]
Thank you very much @manast, I managed to create the queue as I wanted with bullmq.
Congratulations for the excellent work, your lib will help me a lot.

**Comment by felinto-dev** [comment]
What did you do to solve this issue @bricemaurin ? I am facing the same problem.

**Reply by bricemaurin** [reply]
We applied what @manast proposed and it worked perfectly !
